### PR TITLE
New Desing

### PR DESCRIPTION
**Pull request by nerder**

### CODE REVIEW

File: lib/src/localization/localization.dart [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Localization {
   String get registerClass => getValue(RegisterClass);
   String get unregisterClass => getValue(UnregisterClass);
   String get status => getValue(Status);
-  String get lessons => getValue(Lessons);
+  String get lessons => getValue(Classes);
```
Comment (deathcoder) [inline lib/src/localization/localization.dart]: perchè non cambiamo anche il getter da lessons a classes?

File: lib/src/ui/pages/profile_page.dart [hunk 2]
```diff
@@ -173,6 +160,8 @@ class _ProfileState extends State<ProfilePage> {
         });
   }
 
+  bool _isCurrentUser(User profileUser, UserBloc _userBloc) => profileUser.email == (_userBloc.currentState as UserSuccess).currentUser.email;
+
```
Comment (deathcoder) [inline lib/src/ui/pages/profile_page.dart]: serve il parametro _userBloc?

File: lib/src/ui/components/attendees_counter.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        Text("${_getFullPercentage(attendees.length)} full",
+            style:
+                Theme.of(context).textTheme.display1.apply(fontWeightDelta: 3))
+      ],
+    );
+  }
+
+  //TODO this should be configurable
+  String _getFullPercentage(int numberOfAttendees) =>
+      (numberOfAttendees * 100 / 20).toStringAsFixed(0) + "%";
```
Comment (nerder) [inline lib/src/ui/components/attendees_counter.dart]: extract to global constant?

File: lib/src/ui/components/lesson_card.dart [hunk 4]
```diff
@@ -11,64 +13,90 @@ class LessonCard extends StatelessWidget {
...
   }
 }
 
-String capitalize(String s) => s[0].toUpperCase() + s.substring(1);
\ No newline at end of file
+String _capitalize(String s) => s[0].toUpperCase() + s.substring(1);
+
+//TODO this should be configurable
+double _getFullPercentage(int numberOfAttendees) =>
+    (numberOfAttendees * 100 / 20) / 100;
```
Comment (nerder) [inline lib/src/ui/components/lesson_card.dart]: constant

File: lib/src/ui/components/lesson_card.dart [hunk 5]
```diff
@@ -11,64 +13,90 @@ class LessonCard extends StatelessWidget {
...
+                                      child: UserImage(
+                                        userImage: attendee.imageUrl,
+                                        width: MediaQuery.of(context).size.width * 0.06,
+                                        height: MediaQuery.of(context).size.width * 0.06,
+                                      ),
+                                    ))
+                                ?.toList(),
+                            if (lesson.attendees.length > 5)
+                              Text(
+                                  "${Localization.of(context).andOthers} ${lesson.attendees.length - 5}",
```
Comment (nerder) [inline lib/src/ui/components/lesson_card.dart]: maybe the number of user shown should be in a constant too

File: lib/src/ui/pages/registry_page.dart [hunk 6]
```diff
@@ -48,177 +45,115 @@ class _RegistryState extends State<RegistryPage> {
...
                         AttendeesList(
-                          attendeeList: currentLesson.attendees ?? [],
-                          isOwner: _isOwnerUser(),
+                          isOwner: currentUser.isOwner,
+                          isClassEmpty: _isClassEmpty(currentLesson),
+                          isUserInClass:
+                              _isUserInClass(currentLesson, currentUser),
+                          attendeesList: currentLesson.attendees
+                              .where((attendee) =>
+                                  attendee.email != currentUser.email)
```
Comment (nerder) [inline lib/src/ui/pages/registry_page.dart]: this logic should go inside the component since it's responsibility of the child

**Pull request closed without merging**